rotkehlchen/api/v1/resources.py

```diff
@@ -2835,3 +2836,18 @@ def put(
             tx_hash=tx_hash,
             associated_address=associated_address,
         )
+
+
+class EnsAvatarsResource(BaseMethodView):
+    get_schema = EnsAvatarsSchema()
+
+    @require_loggedin_user()
+    @use_kwargs(get_schema, location='json_and_query')
+    def get(self, ens_name: str) -> Response:
+        # Process the if-match and if-none-match headers so that comparison with etag can be done
+        match_header = flask_request.headers.get('If-Match', None)
+        if not match_header:
+            match_header = flask_request.headers.get('If-None-Match', None)
+        if match_header:
+            match_header = match_header[1:-1]  # remove enclosing quotes
```

Comment:
since this block of code is repeated twice... why not just make it a decorator and reuse in both locations?
What do you think?